### ISSUE TITLE
Clicking with mouse increments settings too fast

### ISSUE THREAD

**Issue by nickedyer** [8bitbubsy/ft2-clone#36]
I'm running Pop_OS! and whenever I click on a setting to increment it up or down it increments far too fast for me to be precise. Settings such as BPM, speed, pattern, anything to do with the GUI is incredibly difficult to get to a specific value. I've tried both the pre-compiled binaries and building it from source myself and they both yield the same result.

**Comment by nickedyer** [comment]
I figured it out, it's a problem with my system and VSync not working correctly. Without it working properly the program was running at ~1000FPS and since the calculation is done based on frames it skipped right over the waiting period and skyrocketed. Toggling the VSync off flag has fixed the issue. Would an implementation of changing the calculation for the buttons to use delta timing rather than frame timing be on the table?

**Comment by 8bitbubsy** [comment]
> Would an implementation of changing the calculation for the buttons to use delta timing rather than frame timing be on the table?
Not really, it would require a massive rewrite... But it would be a good idea, yeah.

**Comment by ruby-R53** [comment]
would it still be a good idea now

**Comment by 8bitbubsy** [comment]
Yes, but I'm still not going to do it, because I prefer for the program to be simple in the way it works with frame-based timing. If it can't achieve ~60Hz vsync, it will use sleep-waiting routines instead to try and achieve it. It works on most of the systems I have tested it on...
